tests/test_flask_util.py

```diff
@@ -0,0 +1,325 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Unit tests for the Flask utilities"""
+
+__author__ = 'jonwayne@google.com (Jon Wayne Parrott)'
+
+import httplib2
+import json
+import unittest
+
+import flask
+import mock
+
+from oauth2client import GOOGLE_AUTH_URI
+from oauth2client import GOOGLE_TOKEN_URI
+from oauth2client import clientsecrets
+from oauth2client.flask_util import OAuth2 as FlaskOAuth2
+from oauth2client.client import OAuth2Credentials
+
+
+class Http2Mock(object):
+    """Mock httplib2.Http for code exchange / refresh"""
+    def __init__(self, status=200, **kwargs):
+        self.status = status
+        self.content = {
+            'access_token': 'foo_access_token',
+            'refresh_token': 'foo_refresh_token',
+            'expires_in': 3600,
+            'extra': 'value',
+        }
+        self.content.update(kwargs)
+
+    def request(self, token_uri, method, body, headers, *args, **kwargs):
+        self.body = body
+        self.headers = headers
+        return (self, json.dumps(self.content).encode('utf-8'))
+
+    def __enter__(self):
+        self.httplib2_orig = httplib2.Http
+        httplib2.Http = self
```

Comment:
``__call__`` returns itself, another (dirty) trick. Makes it much easier to mock when I can't control the construction call.